delfin/drivers/ibm/ds8k/ds8k.py

```diff
@@ -0,0 +1,213 @@
+                        total = volume.get('cap')
+                        used = volume.get('capalloc')
+                        vol_type = constants.VolumeType.THICK if \
+                            volume.get('stgtype') == 'fb' else \
+                            constants.VolumeType.THIN
+                        status = constants.StorageStatus.NORMAL if \
+                            volume.get('state') == 'normal' else\
+                            constants.StorageStatus.ABNORMAL
+                        vol_name = '%s_%s' % (volume.get('name'),
+                                              volume.get('id'))
+                        vol = {
+                            'name': vol_name,
+                            'storage_id': self.storage_id,
+                            'description': '',
+                            'status': status,
+                            'native_volume_id': str(volume.get('id')),
+                            'native_storage_pool_id':
+                                volume.get('pool').get('id'),
+                            'wwn': '',
+                            'type': vol_type,
+                            'total_capacity': int(total),
+                            'used_capacity': int(used),
+                            'free_capacity': int(total) - int(used)
+                        }
+                        volume_list.append(vol)
+        return volume_list
+
+    def list_alerts(self, context, query_para=None):
+        alert_model_list = []
+        alert_list = self.rest_handler.get_rest_info(
+            '/api/v1/events?severity=warning,error')
+        alert_handler.AlertHandler() \
+            .parse_queried_alerts(alert_model_list, alert_list, query_para)
+        return alert_model_list
+
+    def list_controllers(self, context):
+        pass
+
+    def list_ports(self, context):
+        port_list = []
+        port_info = self.rest_handler.get_rest_info('/api/v1/ioports')
+        if port_info:
+            port_data = port_info.get('data', {}).get('ioports')
+            for port in port_data:
+                status = constants.PortHealthStatus.NORMAL if \
+                    port.get('state') == 'online' else\
+                    constants.PortHealthStatus.ABNORMAL
+                if port.get('speed'):
+                    speed = int(port.get('speed').split(' ')[0]) * units.G
+                port_result = {
+                    'name': port.get('loc'),
+                    'storage_id': self.storage_id,
+                    'native_port_id': port.get('id'),
+                    'location': port.get('loc'),
+                    'connection_status':
+                        constants.PortConnectionStatus.CONNECTED,
+                    'health_status': status,
+                    'type': DS8KDriver.PORT_TYPE_MAP.get(
+                        port.get('protocol'), constants.PortType.OTHER),
+                    'logical_type': '',
+                    'speed': speed,
```

Comment:
`speed` might be referenced before assignment.

Reply:
done